cmd/docker-app/merge.go

```diff
@@ -39,6 +40,25 @@ func extraFiles(appname string) ([]string, error) {
 	return res, nil
 }
 
+//handleInPlace returns the operation target path and if it's in-place
+func handleInPlace(app *types.App) (string, bool) {
+	if app.Source == types.AppSourceURL || app.Source == types.AppSourceImage {
+		return internal.DirNameFromAppName(app.Name), false
+	}
+	return app.Path + ".tmp", true
+}
+
+// removeAndRename removes target and rename source into target
+func removeAndRename(source, target string) error {
+	if err := os.RemoveAll(target); err != nil {
+		return errors.Wrap(err, "failed to erase previous application")
+	}
+	if err := os.Rename(source, target); err != nil {
```

Comment:
Do we matter that if the rename failed and then the old application (`target`) is removed? The user will have nothing left 😅 
If we do care, maybe then we can rename the target to a temp directory, rename source to target, and if everything is good then delete the temp directory, otherwise move it back to target?

Reply:
Ok just see that the code was moved and factorized. Maybe in a follow-up then 😄